### ISSUE TITLE
promotheus setup undefinied variable

### ISSUE THREAD

**Issue by Happyfeet01** [spantaleev/matrix-docker-ansible-deploy#907]
Hello,

I would install promotheus. That is maybe the reason why facebook and Signal not working.

![Screenshot_20210225_095938](https://user-images.githubusercontent.com/3295104/109131263-e8b2fd00-7752-11eb-8161-f22f96910d4c.jpg)

But i got this error Message when i try to install promotheus.

`TASK [matrix-prometheus : Ensure Prometheus paths exists] **********************
ok: [matrix.dackelexpress.com] => (item=/matrix/prometheus)
ok: [matrix.dackelexpress.com] => (item=/matrix/prometheus/config)
ok: [matrix.dackelexpress.com] => (item=/matrix/prometheus/data)

TASK [matrix-prometheus : Fetch Matrix Docker network details] *****************
ok: [matrix.dackelexpress.com]

TASK [matrix-prometheus : set_fact] ********************************************
fatal: [matrix.dackelexpress.com]: FAILED! => 
  msg: |-
    The task includes an option with an undefined variable. The error was: 'dict object' has no attribute 'network'
  
    The error appears to have been in '/root/matrix-docker-ansible-deploy/roles/matrix-prometheus/tasks/setup_install.yml': line 33, column 7, but may
    be elsewhere in the file depending on the exact syntax problem.
  
    The offending line appears to be:
  
  
        - set_fact:
          ^ here

PLAY RECAP *********************************************************************
matrix.dackelexpress.com   : ok=170  changed=19   unreachable=0    failed=1`

**Comment by spantaleev** [comment]
We do this there:

```yaml
    - set_fact:
        matrix_prometheus_scraper_node_targets: ["{{ matrix_docker_network_info.network.IPAM.Config[0].Gateway }}:9100"]
```

Looks like the result format is likely different.

You can add some `-debug: var="matrix_docker_network_info"` line above this one and let us know what it says.

It's also helpful to see what your Docker version is (`docker version`). Although it's probably the Ansible version or the version of `python-docker` (`python3-docker`, etc.) that is more relevant here.

**Comment by Happyfeet01** [comment]
`TASK [matrix-prometheus : Fetch Matrix Docker network details] *****************
fatal: [matrix.dackelexpress.com]: FAILED! => changed=false 
  msg: 'Unsupported parameters for (docker_network) module: register Supported parameters include: api_version, appends, cacert_path, cert_path, connected, debug, docker_host, driver, driver_options, force, ipam_driver, ipam_options, key_path, network_name, ssl_version, state, timeout, tls, tls_hostname, tls_verify'`

**Comment by jokey2k** [comment]
Experiencing the same issue. Changing the offending line to 
`matrix_prometheus_scraper_node_targets: ["{{ matrix_docker_network_info.ansible_facts.docker_network.IPAM.Config[0].Gateway }}:9100"]`

makes it work okay again, I'm not an ansible expert on why that is.

Running Debian buster
```
#docker version
Client: Docker Engine - Community
 Version:           20.10.4
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        d3cb89e
 Built:             Thu Feb 25 07:05:24 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.4
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       363e9a8
  Built:            Thu Feb 25 07:03:20 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0# docker version
Client: Docker Engine - Community
 Version:           20.10.4
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        d3cb89e
 Built:             Thu Feb 25 07:05:24 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.4
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       363e9a8
  Built:            Thu Feb 25 07:03:20 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

python-docker/stable,stable,now 3.4.1-4 all [installed]

```
TASK [matrix-prometheus : debug] *****************************************************************************************************************************************
ok: [matrix.digital-family.de] =>
  matrix_docker_network_info:
    ansible_facts:
      docker_network:
        Attachable: false
        ConfigFrom:
          Network: ''
        ConfigOnly: false
        Containers:
          0d346b2886584f22791b17327fc6d90f167cd10c9b4a9f7d920b6a36276a03d2:
            EndpointID: 0a81a3367d3a32cf77e2f945b4053c41fd7820227463c60bee8a15a9d7bbace5
            IPv4Address: 172.21.0.2/16
            IPv6Address: ''
            MacAddress: 02:42:ac:15:00:02
            Name: matrix-mailer
          114fe3ec45ef52355422bb92ef7fca20fa7794a54c08fa45f4dc22a5b67e0384:
            EndpointID: 29f71ed6d5f2ad39f4e5238de9a2473113940cde36e9684e531006b5aee96112
            IPv4Address: 172.21.0.3/16
            IPv6Address: ''
            MacAddress: 02:42:ac:15:00:03
            Name: matrix-synapse
          3a4dce6f28648e56b80f2556c75508fd42e0b2b5894c9621508357c7776e4ff3:
            EndpointID: d792ad0147a03e95c355bf274bcd0000d7ddfc7019e7d425d843ba66b8dc5a80
            IPv4Address: 172.21.0.4/16
            IPv6Address: ''
            MacAddress: 02:42:ac:15:00:04
            Name: matrix-postgres
          3c3d4ee102bd51e756d04e46f7241f2986590f384953111a7a4d79de807d461a:
            EndpointID: 62950f4d623bcaa22b03c727fc88f207df05989e71b28cbcc81b6c29d0fcaf4d
            IPv4Address: 172.21.0.9/16
            IPv6Address: ''
            MacAddress: 02:42:ac:15:00:09
            Name: matrix-nginx-proxy
          422c3f47f585e871e4b692edc6ce435c90497bd4059c82ffd6e975b313784c2c:
            EndpointID: a5a01c2d0ab61e9236eabc46904fb940cbb75c0edf258be37e493c549c0fe315
            IPv4Address: 172.21.0.7/16
            IPv6Address: ''
            MacAddress: 02:42:ac:15:00:07
            Name: matrix-registration
          e2a00d90e4ac4647e6f041162d134b4a292639d2755cf0950e955d715b16b31b:
            EndpointID: 2515a95beb77061ee240f391982b29de880c37f5ac9d5e242c9b1ab99d2975ff
            IPv4Address: 172.21.0.5/16
            IPv6Address: ''
            MacAddress: 02:42:ac:15:00:05
            Name: matrix-synapse-admin
          e3d76e35d3c3e95a6c9b11ddefe525dd4fb0bc3be19f947e67801bb76269906f:
            EndpointID: 99b9da4a93ef4b84e9feedb5e0612b060c54be61227383fc6ebdfabd7ac2f1dc
            IPv4Address: 172.21.0.10/16
            IPv6Address: ''
            MacAddress: 02:42:ac:15:00:0a
            Name: matrix-dimension
          e525d0ae9de290ad5a27fed73fd1dfb8bf87f503477c4f358d9122da063beca9:
            EndpointID: a143b92b28dd7c01d2c24a59c98676bf5f089393a3fd9f2621c9902c6399b9c0
            IPv4Address: 172.21.0.6/16
            IPv6Address: ''
            MacAddress: 02:42:ac:15:00:06
            Name: matrix-client-element
          e74a88a6e79b4bdd8c9bff1d3673c2975dc1f19b7d18e9b77434cfb358dcfb92:
            EndpointID: ac3acc93d55713464c6dbb5a4851c6a8fe90133911d66d6d17ed3df4f7f6f59d
            IPv4Address: 172.21.0.8/16
            IPv6Address: ''
            MacAddress: 02:42:ac:15:00:08
            Name: matrix-ma1sd
        Created: '2021-03-01T00:20:05.277595303+01:00'
        Driver: bridge
        EnableIPv6: false
        IPAM:
          Config:
          - Gateway: 172.21.0.1
            Subnet: 172.21.0.0/16
          Driver: default
          Options: null
        Id: cac34ddb403a9bc92a82306b2ecbbded5460485a89bdf66f318a2a4b8600b1d8
        Ingress: false
        Internal: false
        Labels: {}
        Name: matrix
        Options: {}
        Scope: local
    changed: false
    failed: false
```

**Comment by jaapjansma** [comment]
I am having the same problem. 

```
jaap@matrix:~/matrix-docker-ansible-deploy$ docker version
Client: Docker Engine - Community
 Version:           20.10.2
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        2291f61
 Built:             Mon Dec 28 16:17:34 2020
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.5
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       363e9a8
  Built:            Tue Mar  2 20:15:47 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0


```

**Comment by codedeeply** [comment]
> Experiencing the same issue. Changing the offending line to
> `matrix_prometheus_scraper_node_targets: ["{{ matrix_docker_network_info.ansible_facts.docker_network.IPAM.Config[0].Gateway }}:9100"]`
> 
> makes it work okay again, I'm not an ansible expert on why that is.
> 
> Running Debian buster
> 
> ```
> #docker version
> Client: Docker Engine - Community
>  Version:           20.10.4
>  API version:       1.41
>  Go version:        go1.13.15
>  Git commit:        d3cb89e
>  Built:             Thu Feb 25 07:05:24 2021
>  OS/Arch:           linux/amd64
>  Context:           default
>  Experimental:      true
> 
> Server: Docker Engine - Community
>  Engine:
>   Version:          20.10.4
>   API version:      1.41 (minimum version 1.12)
>   Go version:       go1.13.15
>   Git commit:       363e9a8
>   Built:            Thu Feb 25 07:03:20 2021
>   OS/Arch:          linux/amd64
>   Experimental:     false
>  containerd:
>   Version:          1.4.3
>   GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
>  runc:
>   Version:          1.0.0-rc92
>   GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
>  docker-init:
>   Version:          0.19.0
>   GitCommit:        de40ad0# docker version
> Client: Docker Engine - Community
>  Version:           20.10.4
>  API version:       1.41
>  Go version:        go1.13.15
>  Git commit:        d3cb89e
>  Built:             Thu Feb 25 07:05:24 2021
>  OS/Arch:           linux/amd64
>  Context:           default
>  Experimental:      true
> 
> Server: Docker Engine - Community
>  Engine:
>   Version:          20.10.4
>   API version:      1.41 (minimum version 1.12)
>   Go version:       go1.13.15
>   Git commit:       363e9a8
>   Built:            Thu Feb 25 07:03:20 2021
>   OS/Arch:          linux/amd64
>   Experimental:     false
>  containerd:
>   Version:          1.4.3
>   GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
>  runc:
>   Version:          1.0.0-rc92
>   GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
>  docker-init:
>   Version:          0.19.0
>   GitCommit:        de40ad0
> ```
> 
> python-docker/stable,stable,now 3.4.1-4 all [installed]
> 
> ```
> TASK [matrix-prometheus : debug] *****************************************************************************************************************************************
> ok: [matrix.digital-family.de] =>
>   matrix_docker_network_info:
>     ansible_facts:
>       docker_network:
>         Attachable: false
>         ConfigFrom:
>           Network: ''
>         ConfigOnly: false
>         Containers:
>           0d346b2886584f22791b17327fc6d90f167cd10c9b4a9f7d920b6a36276a03d2:
>             EndpointID: 0a81a3367d3a32cf77e2f945b4053c41fd7820227463c60bee8a15a9d7bbace5
>             IPv4Address: 172.21.0.2/16
>             IPv6Address: ''
>             MacAddress: 02:42:ac:15:00:02
>             Name: matrix-mailer
>           114fe3ec45ef52355422bb92ef7fca20fa7794a54c08fa45f4dc22a5b67e0384:
>             EndpointID: 29f71ed6d5f2ad39f4e5238de9a2473113940cde36e9684e531006b5aee96112
>             IPv4Address: 172.21.0.3/16
>             IPv6Address: ''
>             MacAddress: 02:42:ac:15:00:03
>             Name: matrix-synapse
>           3a4dce6f28648e56b80f2556c75508fd42e0b2b5894c9621508357c7776e4ff3:
>             EndpointID: d792ad0147a03e95c355bf274bcd0000d7ddfc7019e7d425d843ba66b8dc5a80
>             IPv4Address: 172.21.0.4/16
>             IPv6Address: ''
>             MacAddress: 02:42:ac:15:00:04
>             Name: matrix-postgres
>           3c3d4ee102bd51e756d04e46f7241f2986590f384953111a7a4d79de807d461a:
>             EndpointID: 62950f4d623bcaa22b03c727fc88f207df05989e71b28cbcc81b6c29d0fcaf4d
>             IPv4Address: 172.21.0.9/16
>             IPv6Address: ''
>             MacAddress: 02:42:ac:15:00:09
>             Name: matrix-nginx-proxy
>           422c3f47f585e871e4b692edc6ce435c90497bd4059c82ffd6e975b313784c2c:
>             EndpointID: a5a01c2d0ab61e9236eabc46904fb940cbb75c0edf258be37e493c549c0fe315
>             IPv4Address: 172.21.0.7/16
>             IPv6Address: ''
>             MacAddress: 02:42:ac:15:00:07
>             Name: matrix-registration
>           e2a00d90e4ac4647e6f041162d134b4a292639d2755cf0950e955d715b16b31b:
>             EndpointID: 2515a95beb77061ee240f391982b29de880c37f5ac9d5e242c9b1ab99d2975ff
>             IPv4Address: 172.21.0.5/16
>             IPv6Address: ''
>             MacAddress: 02:42:ac:15:00:05
>             Name: matrix-synapse-admin
>           e3d76e35d3c3e95a6c9b11ddefe525dd4fb0bc3be19f947e67801bb76269906f:
>             EndpointID: 99b9da4a93ef4b84e9feedb5e0612b060c54be61227383fc6ebdfabd7ac2f1dc
>             IPv4Address: 172.21.0.10/16
>             IPv6Address: ''
>             MacAddress: 02:42:ac:15:00:0a
>             Name: matrix-dimension
>           e525d0ae9de290ad5a27fed73fd1dfb8bf87f503477c4f358d9122da063beca9:
>             EndpointID: a143b92b28dd7c01d2c24a59c98676bf5f089393a3fd9f2621c9902c6399b9c0
>             IPv4Address: 172.21.0.6/16
>             IPv6Address: ''
>             MacAddress: 02:42:ac:15:00:06
>             Name: matrix-client-element
>           e74a88a6e79b4bdd8c9bff1d3673c2975dc1f19b7d18e9b77434cfb358dcfb92:
>             EndpointID: ac3acc93d55713464c6dbb5a4851c6a8fe90133911d66d6d17ed3df4f7f6f59d
>             IPv4Address: 172.21.0.8/16
>             IPv6Address: ''
>             MacAddress: 02:42:ac:15:00:08
>             Name: matrix-ma1sd
>         Created: '2021-03-01T00:20:05.277595303+01:00'
>         Driver: bridge
>         EnableIPv6: false
>         IPAM:
>           Config:
>           - Gateway: 172.21.0.1
>             Subnet: 172.21.0.0/16
>           Driver: default
>           Options: null
>         Id: cac34ddb403a9bc92a82306b2ecbbded5460485a89bdf66f318a2a4b8600b1d8
>         Ingress: false
>         Internal: false
>         Labels: {}
>         Name: matrix
>         Options: {}
>         Scope: local
>     changed: false
>     failed: false
> ```

This fixed the same issue for me on Debian Buster.

**Comment by thedanbob** [comment]
Comparing https://docs.ansible.com/ansible/2.7/modules/docker_network_module.html and https://docs.ansible.com/ansible/2.8/modules/docker_network_module.html it looks like the network dict was added in ansible 2.8. Debian Buster is still on 2.7.7.

**Comment by spantaleev** [comment]
I've tried to fix it in 68ca81c. Let me know if it works for you!

**Comment by thedanbob** [comment]
I saw you reverted the fix, it actually worked well once I wrapped the default filter with parenthesis. See #1006
